crates/hotshot/src/lib.rs

```diff
@@ -166,6 +166,9 @@ pub struct SystemContext<TYPES: NodeType, I: NodeImplementation<TYPES>> {
 
     /// a potential upgrade certificate that has been decided on by the consensus tasks.
     pub decided_upgrade_certificate: Arc<RwLock<Option<UpgradeCertificate<TYPES>>>>,
+
+    /// Reference to the AuctionResults type for acquiring solver results.
+    pub auction_results: Arc<RwLock<I::AuctionResults>>,
```

Comment:
I wonder if this can just be `Arc` without the `RwLock`: it feels like all the operations on the auction results should be immutable

Reply:
Make sense, swapping.